game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/AvailableSupports.java

```diff
@@ -0,0 +1,143 @@
+package games.strategy.triplea.delegate.power.calculator;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.attachments.UnitSupportAttachment;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import org.triplea.java.collections.IntegerMap;
+
+/**
+ * Tracks the available support that a collection of units can give to other units.
+ *
+ * <p>Once a support is used, it will no longer be available for other units to use.
+ */
+@Data
+@Builder(access = AccessLevel.PRIVATE)
+@AllArgsConstructor
+class AvailableSupports {
+
+  static final AvailableSupports EMPTY_RESULT =
+      AvailableSupports.builder()
+          .supportRules(new HashMap<>())
+          .supportUnits(new HashMap<>())
+          .build();
+
+  final Map<UnitSupportAttachment.BonusType, List<UnitSupportAttachment>> supportRules;
+  final Map<UnitSupportAttachment, IntegerMap<Unit>> supportUnits;
+
+  /** Sorts 'supportsAvailable' lists based on unit support attachment rules. */
+  static AvailableSupports getSortedSupport(final SupportCalculator supportCalculator) {
+    final AvailableSupports supportCalculationResult = getSupport(supportCalculator);
+
+    final SupportRuleSort supportRuleSort =
+        SupportRuleSort.builder()
+            .defense(supportCalculator.isDefence())
+            .friendly(supportCalculator.isAllies())
+            .roll(UnitSupportAttachment::getRoll)
+            .strength(UnitSupportAttachment::getStrength)
+            .build();
+    supportCalculationResult
+        .getSupportRules()
+        .forEach((bonusType, unitSupportAttachment) -> unitSupportAttachment.sort(supportRuleSort));
+    return supportCalculationResult;
+  }
+
+  static AvailableSupports getSupport(final SupportCalculator supportCalculator) {
+    return builder()
+        .supportRules(supportCalculator.getSupportRules())
+        .supportUnits(supportCalculator.getSupportUnits())
+        .build();
+  }
+
+  /** Constructs a filtered version of this */
+  AvailableSupports filter(final Predicate<UnitSupportAttachment> ruleFilter) {
+
+    final Map<UnitSupportAttachment.BonusType, List<UnitSupportAttachment>> supportRules =
```

Comment:
Any thoughts to use a guava multimap here? That datastructure is equivalent to a `Map<X, Collection<Y>>`

After instantiating the map, you should be able to simplify and write:

```
final var supportRules =     HashMultimap.create();
entry.getValue().stream().filter(ruleFilter).forEach(supportRules::put)
```

Reply:
I think this method might go away similar to the `Predicate` method that you commented on earlier.